File: Data Structures & Algorithms/meeting-schedule-ii/submission-4.cpp
/**
 * Definition of Interval:
 * class Interval {
 * public:
 *     int start, end;
 *     Interval(int start, int end) {
 *         this->start = start;
 *         this->end = end;
 *     }
 * }
 */

class Solution {
public:
    int minMeetingRooms(vector<Interval>& intervals) {
        vector<int> start;
        vector<int> ending;
        for(Interval i: intervals){
            start.push_back(i.start);
            ending.push_back(i.end);
        }
        sort(start.begin(), start.end());
        sort(ending.begin(), ending.end());
        int ans = 0;
        int res = 0;
        int i = 0;
        int j = 0;
        while(i < start.size() && j < ending.size()){
            if(start[i] < ending[j]){
                res++;
                ++i;
            }else{
                res--;
                ++j;
            }
            ans = max(ans, res);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()
        for i in nums:
            s.add(i)
        res = 0
        for num in s:
            if num - 1 not in s:
                counter = 1
                while num + counter in s:
                    counter += 1
                res = max(res, counter)
        return res
